prereise/gather/demanddata/transportation_electrification/charging.py

```diff
@@ -0,0 +1,114 @@
+def get_charging_power(power, trip_strategy, location_strategy, kwh, trip_data):
+    """Determines the charging power. Returns either param power or 0.
+
+    :param int power: charger power, EVSE kW.
+    :param int trip_strategy: a flag that determines which trips to consider, 1-anytrip number, 2-last trip.
+    :param int location_strategy: where the vehicle can charge-1, 2, 3, 4, or 5; 
+        1-home only, 2-home and work related, 3-anywhere if possibile, 
+        4-home and school only, 5-home and work and school.
+    :param float kwh: kwhmi * veh_range, amount of energy needed to charge vehicle.
+    :param pandas.Series trip_data: Row of DataFrame that has data for the trip we are calculating for.
+    :return: (*float*) -- charging power.
+    """
+    charging_power = power
+
+    dwelling = int(trip_data["Dwell time (hour decimal)"] > 0.2)
+    location = int(get_location(location_strategy, trip_data["why to"]))
+
+    trip = int(
+        consider_trip_number(
+            trip_strategy, trip_data["total vehicle trips"], trip_data["trip number"]
+        )
+    )
+
+    battery = int(get_battery_SOC(trip_data["trip end battery charge"], kwh))
+
+    if (dwelling * location * trip * battery != 1) or not (
+        (dwelling * location == 1)
+        or (trip_data["total vehicle trips"] == trip_data["trip number"])
+    ):
```

Comment:
In the original matlab code, one of the if-statements was to check if `dwelling` and `location` are equal to 1 while being on the last trip, then it would return the charging power. However, for this version, all the if-statements were consolidated (hence, the first condition and the second big condition) and we took their inverse. If they're not satisfied, then it returns 0.

Thinking more about it, it does seem redundant because whether or not it's on the last trip, if `dwelling` and `location` are 0, then it would be handled by the first condition as you mentioned. So I decided to remove the second portion and update it per your suggestion.